src/persist/persist.ts

```diff
@@ -81,18 +81,24 @@ async function writeFixedChunks(
   mainHeadHash: Hash,
   clientID: string,
 ) {
-  await perdag.withWrite(async dagWrite => {
+  const tempHeadName = `${clientID}-fixed-chunks`;
+
+  const clients = await perdag.withWrite(async dagWrite => {
     const ps: Promise<void>[] = [];
     for (const chunk of fixedChunks.values()) {
       ps.push(dagWrite.putChunk(chunk));
     }
-    const client: Client = {
-      heartbeatTimestampMs: Date.now(),
-      headHash: mainHeadHash,
-    };
-    // We need to set a head here or the chunks will be GC'd
-    ps.push(setClient(clientID, client, dagWrite));
+
+    // We set a head here to make sure the chunks are not garbage collected.
+    // This head will be removed when the clients head is updated.
+    ps.push(dagWrite.setHead(tempHeadName, mainHeadHash));
```

Comment:
If this tab crashes or is frozen before updateClients completes this temp head will never be cleaned up.

We should update client-gc.ts, so that when it gc's a client it also cleans up any lingering temp heads for that client.